modules/claims_api/app/controllers/concerns/claims_api/disability_compensation_validations.rb

```diff
@@ -411,17 +414,30 @@ def validate_form_526_disability_secondary_disabilities!
     end
 
     def validate_form_526_disability_classification_code!
-      return if (form_attributes['disabilities'].pluck('classificationCode') - [nil]).blank?
-
       form_attributes['disabilities'].each do |disability|
-        next if disability['classificationCode'].blank?
-        next if bgs_classification_ids.include?(disability['classificationCode'])
-
-        raise ::Common::Exceptions::InvalidFieldValue.new('disabilities.classificationCode',
-                                                          disability['classificationCode'])
+        classification_code = disability['classificationCode']
+        next if classification_code.nil? || classification_code.blank?
+
+        if bgs_classification_ids.include?(classification_code)
+          validate_form_526_disability_classification_code_end_date!(classification_code)
+        else
+          raise ::Common::Exceptions::InvalidFieldValue.new('disabilities.classificationCode',
+                                                            classification_code)
+        end
       end
     end
 
+    def validate_form_526_disability_classification_code_end_date!(classification_code)
+      brd_disability = brd_disabilities.find { |d| d[:id] == classification_code.to_i }
+      end_date_time = brd_disability[:endDateTime] if brd_disability
+
+      return if end_date_time.nil?
+
+      return if Date.parse(end_date_time) >= Time.zone.today
+
+      raise ::Common::Exceptions::InvalidFieldValue.new('disabilities.classificationCode', classification_code)
+    end
+
     def bgs_classification_ids
       return @bgs_classification_ids if @bgs_classification_ids.present?
```

Comment:
ooh. I think we want to move away from BGS and go to BRD for this. I think that's the intent with the require statement but we're still hitting BGS here. getting confirmation from @klwatanabe

Reply:
🤷 let us know!

Reply:
yeah, sounds like we wanna stay with BGS just for fear of changes breaking our Sandbox environment due to the way BRD is mocked and BGS is not 😢 - so this will not move to BRD without further discovery / discussion.